data-serving/data-service/src/model/revision-metadata.ts

```diff
@@ -11,20 +23,29 @@ export const revisionMetadataSchema = new mongoose.Schema({
         },
         required: 'Enter a revision id',
     },
-    moderator: {
-        type: String,
-        required: 'Enter a revision moderator id',
+    creationMetadata: {
+        type: editMetadataSchema,
+        required: 'Enter creation metadata',
     },
-    date: {
-        ...dateFieldInfo,
-        required: 'Enter a revision date',
+    updateMetadata: {
+        type: editMetadataSchema,
+        required: [
+            function (this: RevisionMetadataDocument): boolean {
+                return this.revisionNumber > 0 && !this.updateMetadata;
```

Comment:
Is the second part of this conditional necessary? Isn't this function the conditions under which the data is required?

Reply:
ugh no I ran into a weird bug and was just throwing stuff against the wall and forgot to remove. thanks!